client/store/current-user/actions.js

```diff
@@ -0,0 +1,33 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { dispatch } from '@wordpress/data';
+import apiRequest from '@wordpress/api-request';
+
+export default {
+	setCurrentUser( user ) {
+		return {
+			type: 'SET_CURRENT_USER',
+			user,
+		};
+	},
+	updateWoocommerceUserMeta( meta ) {
+		return async () => {
+			const data = {
+				woocommerce_meta: { ...meta },
+			};
+			try {
+				const updatedUser = await apiRequest( {
```

Comment:
It seems like core doesn't use this approach of putting api requests inside actions, rather [their actions are all plain objects](https://github.com/WordPress/gutenberg/blob/master/packages/core-data/src/actions.js), and it looks like instead they have [a middleware for effects](https://github.com/WordPress/gutenberg/blob/master/packages/editor/src/store/effects.js), where post/delete/put requests happen. [This `trashPost` function](https://github.com/WordPress/gutenberg/blob/master/packages/editor/src/store/effects/posts.js#L268), which is triggered [on the TRASH_POST action](https://github.com/WordPress/gutenberg/blob/master/packages/editor/src/store/effects.js#L87) seems like a relatively simple example, though I'm not sure how we can hook into that (maybe we also set up [a middlewares file](https://github.com/WordPress/gutenberg/blob/master/packages/editor/src/store/middlewares.js)). `edit-post` also has this `effects` pattern, and I don't think anything else in gutenberg updates data.